File: dsa-problems/Queues/solutions/cpp/QUE-001-campus-service-line.cpp
#include <iostream>
#include <vector>
#include <string>
#include <queue>

using namespace std;

class Solution {
public:
    vector<string> processCommands(const vector<vector<string>>& commands) {
        queue<string> q;
        vector<string> result;
        
        for (const auto& cmd : commands) {
            const string& op = cmd[0];
            if (op == "ENQUEUE") {
                q.push(cmd[1]);
            } else if (op == "DEQUEUE") {
                if (q.empty()) {
                    result.push_back("EMPTY");
                } else {
                    result.push_back(q.front());
                    q.pop();
                }
            } else if (op == "FRONT") {
                if (q.empty()) {
                    result.push_back("EMPTY");
                } else {
                    result.push_back(q.front());
                }
            }
        }
        return result;
    }
};

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int m;
    if (cin >> m) {
        vector<vector<string>> commands;
        commands.reserve(m);
        for (int i = 0; i < m; i++) {
            string op;
            cin >> op;
            if (op == "ENQUEUE") {
                string val;
                cin >> val;
                commands.push_back({op, val});
            } else {
                commands.push_back({op});
            }
        }
    
        Solution solution;
        vector<string> result = solution.processCommands(commands);
        for (const string& line : result) {
            cout << line << "\n";
        }
    }
    return 0;
}

File: dsa-problems/Queues/solutions/java/QUE-001-campus-service-line.java
import java.util.*;

class Solution {
    public List<String> processCommands(List<String[]> commands) {
        List<String> result = new ArrayList<>();
        Queue<String> queue = new LinkedList<>();

        for (String[] cmd : commands) {
            String op = cmd[0];
            if (op.equals("ENQUEUE")) {
                queue.offer(cmd[1]);
            } else if (op.equals("DEQUEUE")) {
                if (queue.isEmpty()) {
                    result.add("EMPTY");
                } else {
                    result.add(queue.poll());
                }
            } else if (op.equals("FRONT")) {
                if (queue.isEmpty()) {
                    result.add("EMPTY");
                } else {
                    result.add(queue.peek());
                }
            }
        }
        return result;
    }
}

class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        if (sc.hasNextInt()) {
            int m = sc.nextInt();
            List<String[]> commands = new ArrayList<>();
    
            for (int i = 0; i < m; i++) {
                String op = sc.next();
                if (op.equals("ENQUEUE")) {
                    String x = sc.next();
                    commands.add(new String[]{op, x});
                } else {
                    commands.add(new String[]{op});
                }
            }
    
            Solution solution = new Solution();
            List<String> result = solution.processCommands(commands);
            for (String line : result) {
                System.out.println(line);
            }
        }
        sc.close();
    }
}
